gui/saveload-dialog.cpp

```diff
@@ -482,29 +487,37 @@ void SaveLoadChooserSimple::reflowLayout() {
 			textLines++;
 		if (textLines > 0)
 			textLines++; // add a line of padding at the bottom
-
-		_container->resize(x, y, w, h + (kLineHeight * textLines));
-		_gfxWidget->resize(thumbX, thumbY, thumbW, thumbH);
-
+	
+		
+		if (_thumbnailSupport) {
+			_gfxWidget->resize(thumbX, thumbY, thumbW, thumbH);	
+			_gfxWidget->setVisible(true);
+		}
+		else { // choose sensible values for displaying playtime and date/time when a thumbnail is not being used
+			thumbH = 0;
+			thumbY = y;
+			_gfxWidget->setVisible(false);
+			h = kLineHeight;
+		}
+		
 		int height = thumbY + thumbH + kLineHeight;
 
 		if (_saveDateSupport) {
-			_date->resize(thumbX, height, kThumbnailWidth, kLineHeight);
+			_date->resize(thumbX, height, thumbW, kLineHeight);
 			height += kLineHeight;
-			_time->resize(thumbX, height, kThumbnailWidth, kLineHeight);
+			_time->resize(thumbX, height, thumbW, kLineHeight);
 			height += kLineHeight;
+			_date->setVisible(_saveDateSupport);
+			_time->setVisible(_saveDateSupport);
```

Comment:
By moving those inside the if block aren't you now missing a setVisible(false) when _saveDateSupport is false?